packages/components/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,263 @@
+import { Icon } from '../Icon/Icon'
+import { Spacer } from '../Spacer/Spacer'
+import { useTheme } from '../../utils'
+
+export type CheckboxProps = {
+  /** Primary text for checkbox */
+  label: string
+  /** OnPress function. Pass a function that alters `checked` state */
+  onPress: () => void
+  /** Optional boolean that determines whether checkbox appears checked */
+  checked?: boolean
+  /** Optional description that appears below label */
+  description?: string
+  /** Optional accessibility override for description text  */
+  descriptionA11y?: string
+  /** Optional hint text. Appears below header */
+  hint?: string
+  /** Optional accessibility override for hint text  */
+  hintA11y?: string
+  /** Optional error text. Non-null value styles checkbox in error state */
+  error?: string
+  /** Optional accessibility override for error text  */
+  errorA11y?: string
+  /** Optional header text */
+  header?: string
+  /** Optional accessibility override for header  */
+  headerA11y?: string
+  /** Optional indeterminate boolean. Overrides checked styling */
+  indeterminate?: boolean
+  /** Optional accessibility override for label  */
+  labelA11y?: string
+  /** Optional required boolean that denotes a label as (*Required) */
+  required?: boolean
+  /** Optional tile styling */
+  tile?: boolean
+}
+
+export const Checkbox: FC<CheckboxProps> = ({
+  checked,
+  label,
+  labelA11y,
+  description,
+  descriptionA11y,
+  error,
+  errorA11y,
+  header,
+  headerA11y,
+  hint,
+  hintA11y,
+  indeterminate,
+  onPress,
+  required,
+  tile,
+}) => {
+  const { t } = useTranslation()
+  const theme = useTheme()
+
+  const baseTextStyle = {
+    fontFamily: 'SourceSansPro-Regular',
+    color: theme.vadsColorForegroundDefault,
+  }
+
+  /**
+   * Container styling
+   */
+
+  let containerStyle: ViewStyle = {
+    width: '100%',
+  }
+
+  if (error) {
+    containerStyle = {
+      ...containerStyle,
+      borderLeftWidth: spacing.vadsSpace2xs,
+      borderColor: theme.vadsColorFormsBorderError,
+      paddingLeft: spacing.vadsSpaceLg,
+    }
+  }
+
+  /**
+   * Pressable styling
+   */
+
+  const pressableBaseStyle: StyleProp<ViewStyle> = {
+    width: '100%',
+    flexDirection: 'row',
+    alignItems: 'flex-start',
+  }
+
+  const tileStyle: ViewStyle = {
+    ...pressableBaseStyle,
+    borderWidth: 2,
+    borderRadius: spacing.vadsSpace2xs,
+    paddingTop: spacing.vadsSpaceSm,
+    paddingBottom: spacing.vadsSpaceSm,
+    paddingLeft: spacing.vadsSpaceSm,
+    paddingRight: spacing.vadsSpaceMd,
+    borderColor: checked
+      ? theme.vadsColorFormsBorderActive
+      : theme.vadsColorFormsBorderSubtle,
+    backgroundColor: checked
+      ? theme.vadsColorFormsSurfaceActive
+      : theme.vadsColorSurfaceDefault,
+  }
+
+  /**
+   * Icon
+   */
+
+  const iconViewStyle: ViewStyle = {
+    alignItems: 'center',
+    justifyContent: 'center',
+    paddingTop: 3,
+    paddingBottom: 3,
+  }
```

Comment:
I didn't check, but think this will break with font scaling to maintain the checkbox centered with the text. Recommend following what we do in Alert/Link/Snackbar (including updates in `main`, but not yet your branch to depend on the font).